dbms/src/Columns/ColumnArray.cpp

```diff
@@ -178,11 +178,16 @@ StringRef ColumnArray::serializeValueIntoArena(size_t n, Arena & arena, char con
     char * pos = arena.allocContinue(sizeof(array_size), begin);
     memcpy(pos, &array_size, sizeof(array_size));
 
-    size_t values_size = 0;
+    StringRef res(pos, sizeof(array_size));
+
     for (size_t i = 0; i < array_size; ++i)
-        values_size += getData().serializeValueIntoArena(offset + i, arena, begin).size;
+    {
+        auto value_ref = getData().serializeValueIntoArena(offset + i, arena, begin);
+        res.data = value_ref.data - res.size;
+        res.size += value_ref.size;
+    }
 
-    return StringRef(begin, sizeof(array_size) + values_size);
+    return res;
```

Comment:
So now we return StringRef to the serialized value, but apparently before we returned some wrong StringRef where data points to the start of chunk, and size is the size of serialized value? How did that even work?

Reply:
It wasn't important because in most cases data ptr was not used at all.
Typical case - call method several times, and just use `begin` and total size.